src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,339 @@
+                                           order='F')
+                    data = np.dstack((data, new_frame)) if data.size else new_frame
+
+            return data
+
+
+        else:
+            
+            print('frame_count.Regions is not. Please retry.')
+            print(frame_count.Frames)
+        
+
+    def close(self):
+        """
+        Similar to finalize as seen a few lines below. 
+        """
+        self.automation.Dispose()
+        print('Closed AddInProcess.exe')
+
+
+class Spectrometer():
+    GRATINGS = ['[500nm,600][0][0]', '[1.2um,300][1][0]',
+                '[500nm,150][2][0]']  # TODO: make it so that it automatically scrapes the grating settings
+
+    def initialize(self):
+        """
+        Sets up LightField and loads an empty experiment called 
+        "LF_Control that we use for automation purposes.
+        """
+        self.light = LightfieldApp(True)
+        self.light.load_experiment('LF_Control')  
+
+    #TODO:Need to create a setting that automatically deletes all saved scans on pc.
+    def finalize(self):
+        """
+        Closes the Lightfield application without saving the settings.
+        """
+        self.light.close()
+
+    @property
+    def center_wavelength(self):
+        """
+        Returns the center wavelength in nanometers.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingCenterWavelength)
+
+    def get_wavelengths(self):
+        """
+        Returns the wavelength calibration for a single frame.
+        """
+        size = self.light.experiment.SystemColumnCalibration.get_Length()
+
+        result = np.empty(size)
+
+        for i in range(size):
+            result[i] = self.light.experiment.SystemColumnCalibration[i]
+
+        return result
+
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers):
+        """
+        Sets the spectrometer center wavelength to nanometers.
+        """
+        # this avoids bug where if step and glue is selected, doesn't allow setting center wavelength
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+
+        return self.light.set(lf.AddIns.SpectrometerSettings.GratingCenterWavelength, nanometers)
+
+    @property
+    def grating(self):
+        """
+        Returns the current grating.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
+
+    @grating.setter
+    def grating(self, grating):
+        """
+        Sets the current grating to be the one specified by parameter grating.
+        """
+        # TODO: figure out the format for setting this
+
+        print('still need to figure out the format for this')
+
+    @property
+    def gratings(self):
+        """
+        Returns a list of all installed gratings.
+        """
+        break_down = False
+
+        if break_down:
+
+            import re
+
+            for g in GRATINGS:
```

Comment:
I'm a little surprised this works, actually. Shouldn't it be `Spectrometer.GRATINGS`?

Reply:
I think you are right, it's supposed to be Spectrometer.gratings. Also I don't think it works since this should be the line that returns the gratings: 
```python
@property
    def grating(self):
        """
        Returns the current grating.
        """
        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
```